test/libevmasm/Assembler.cpp

```diff
@@ -100,15 +103,19 @@ BOOST_AUTO_TEST_CASE(all_assembly_items)
 
 	checkCompilation(_assembly);
 
+	ErrorList errors;
+	ErrorReporter errorReporter(errors);
+
 	BOOST_CHECK_EQUAL(
-		_assembly.assemble().toHex(),
+		_assembly.assemble(errorReporter).toHex(),
 		"5b6001600220606f73__$bf005014d9d0f534b8fcb268bd84c491a2$__"
 		"60005660676022604573000000000000000000000000000000000000000050"
 		"60028060015250"
 		"00fe"
 		"7f0000000000000000000000000000000000000000000000000000000000000000"
 		"fe010203044266eeaa"
 	);
+	BOOST_CHECK(!errorReporter.hasErrors());
```

Comment:
I thought about making class `ErrorAsserter: public ErrorReporter` which would override `error` function and throw on any error. Then we could write just `_assembly.assemble(ErrorAsserter()).toHex()` and avoid boilder plate code in several places.